src/MigrationTools.Clients.AzureDevops.ObjectModel/Enrichers/TfsWorkItemEmbededLinkEnricher.cs

```diff
@@ -0,0 +1,145 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text.RegularExpressions;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using Microsoft.TeamFoundation.Client;
+using Microsoft.TeamFoundation.Framework.Client;
+using Microsoft.TeamFoundation.Framework.Common;
+using Microsoft.TeamFoundation.WorkItemTracking.Client;
+using MigrationTools.DataContracts;
+using MigrationTools.Processors;
+
+namespace MigrationTools.Enrichers
+{
+    public class TfsWorkItemEmbededLinkEnricher : WorkItemProcessorEnricher
+    {
+        private const string RegexPatternLinkAnchorTag = "<a[^>].*?(?:href=\"(?<href>[^\"]*)\".*?|(?<version>data-vss-mention=\"[^\"]*\").*?)*>(?<value>.*?)<\\/a?>";
+        private const string RegexPatternWorkItemUrl = "http[s]*://.*?/_workitems/edit/(?<id>\\d+)";
+        private readonly Lazy<List<TeamFoundationIdentity>> _targetTeamFoundationIdentities;
+
+        private IMigrationEngine Engine;
+        
+        public TfsWorkItemEmbededLinkEnricher(IServiceProvider services, ILogger<TfsWorkItemEmbededLinkEnricher> logger)
+            : base(services, logger)
+        {
+            Engine = services.GetRequiredService<IMigrationEngine>();
+            _targetTeamFoundationIdentities = new Lazy<List<TeamFoundationIdentity>>(() =>
+            {
+                TfsTeamService teamService = Engine.Target.GetService<TfsTeamService>();
+                TfsConnection connection = (TfsConnection)Engine.Target.InternalCollection;
+
+                var identityService = Engine.Target.GetService<IIdentityManagementService>();
+                var tfi = identityService.ReadIdentity(IdentitySearchFactor.General, "Project Collection Valid Users", MembershipQuery.Expanded, ReadIdentityOptions.None);
+                return identityService.ReadIdentities(tfi.Members, MembershipQuery.None, ReadIdentityOptions.None).ToList();
+            });
+        }
+
+        [Obsolete]
+        public override void Configure(bool save = false, bool filterWorkItemsThatAlreadyExistInTarget = true)
+        {
+            throw new NotImplementedException();
+        }
+
+        [Obsolete]
+        public override int Enrich(WorkItemData sourceWorkItem, WorkItemData targetWorkItem)
+        {
+            string oldTfsurl = Engine.Source.Config.AsTeamProjectConfig().Collection.ToString();
+            string newTfsurl = Engine.Target.Config.AsTeamProjectConfig().Collection.ToString();
+
+            var logTypeName = nameof(TfsWorkItemEmbededLinkEnricher);
+            Log.LogInformation($"{logTypeName}: Fixing embedded mention links on target work item {targetWorkItem.Id} from {oldTfsurl} to {newTfsurl}");
```

Comment:
All Logging should be in the format `"My message that contains {item} and {item2}", item, itesm2`! Do not use `$"My message that contains {item} and {item2}"` to pass text into the log strings as this disables [Serilog](https://serilog.net/)'s ability to pass that data as telemetry to Application Insights and for log highlighting.

Reply:
Updated..